src/components/App/App.js

```diff
@@ -23,18 +16,31 @@ export default class App extends React.Component {
   }
 
   getLocals() {
+    const {
+      onSearchChange,
+      onLogoClick,
+      state: {
+        searchValue,
+        resultsValue
+      }
+    } = this;
+
     return {
-      onSearchChange : this.onSearchChange,
-      searchValue : this.state.searchValue,
-      onLogoClick: this.onLogoClick,
-      resultsValue: this.state.resultsValue
+      onSearchChange,
+      searchValue,
+      onLogoClick,
+      resultsValue
     };
   }
 
   onSearchChange = value => {
     //This is used for testing purpose only
-    const results = (value === 'reactjs' ? fakeResults : []);
-    this.setState({ searchValue : value, resultsValue : results });
+    //const results = (value === 'reactjs' ? fakeResults : []);
+
+    getRepos(value)
+      .then(res => {
+        this.setState({ searchValue : value, resultsValue : res });
+      }).catch(::console.error);
```

Comment:
we should actually handle the error and notify the user. Not in scope for this issue though, so tomorrow